projects/objects/road/protos/Roundabout.proto

```diff
@@ -150,60 +148,59 @@ PROTO Roundabout [
-          centralPoint = wbgeometry.findclosest2Dpointinarray(centralPoint, outerPoints)
-          local point = { x = outerRadius * math.cos(angle+theta2), y = outerRadius * math.sin(angle+theta2) }
-          point = wbgeometry.findclosest2Dpointinarray(point, outerPoints)
-          local centralIndex = 0
-          local pointIndex = 0
-          local pointsNumber = wbcore.tablelength(outerPoints)
-          for i = 1, pointsNumber do
-            if wbvector2.equal(outerPoints[i], point) then
-              pointIndex = i
-            elseif wbvector2.equal(outerPoints[i], centralPoint) then
-              centralIndex = i
-            end
-          end
-          numberOfIndex = math.abs(pointIndex - centralIndex)
-        }%
+      %< for (let i = 0; i <= roadNumber - 1; ++i) { >%
+        %<
+          let angle         = 2 * Math.PI * i / roadNumber + Math.PI / roadNumber;
+          let x             = outerRadius * Math.cos(angle);
+          let z             = outerRadius * Math.sin(angle);
+          let rotationAngle = -2 * Math.PI * (i + 0.5) / roadNumber + Math.PI * 0.5;
+          let theta = Math.atan2(roadsWidth * 0.5, outerRadius);
+          let d = roadsWidth * 0.5 - Math.sin(theta) * outerRadius;
+          let theta2 = Math.atan2(roadsWidth * 0.5 + 5 * d, outerRadius);
+          let centralPoint = {x: x, y: z};
+          centralPoint = wbgeometry.findClosestPoint2InArray(centralPoint, outerPoints);
+          let point = {x: outerRadius * Math.cos(angle+theta2), y: outerRadius * Math.sin(angle+theta2)};
+          point = wbgeometry.findClosestPoint2InArray(point, outerPoints);
+          let centralIndex = 0;
+          let pointIndex = 0;
+          let pointsNumber = outerPoints.length;
+          for (let i = 0; i < pointsNumber; ++i) {
+            if (wbvector2.equal(outerPoints[i], point)) {
+              pointIndex = i;
+            } else if (wbvector2.equal(outerPoints[i], centralPoint)) {
+              centralIndex = i;
+            }
+          }
+          let numberOfIndex = Math.abs(pointIndex - centralIndex);
+        >%
         # The border of the road
-        %{
-          local roadLength = startRoadsLength[1]
-          if useIndependentStartRoadLength then
-            roadLength = startRoadsLength[i + 1]
-          end
-        }%
-        %{ if roadLength > 0 then }%
+        %<
+          let roadLength = startRoadsLength[0];
+          if (useIndependentStartRoadLength)
+            roadLength = startRoadsLength[i + 1];
+        >%
+        %< if (roadLength > 0) { >%
         Road {
-          translation %{= centralPoint.x }% 0 %{= centralPoint.y }%
-          rotation 0 1 0 %{= rotationAngle }%
-          name "road border %{=i}%"
-          width %{= roadsWidth }%
+          translation %<= centralPoint.x >% 0 %<= centralPoint.y >%
+          rotation 0 1 0 %<= rotationAngle >%
+          name "road border %<=i>%"
+          width %<= roadsWidth >%
```

Comment:
```suggestion
          name "road border %<= i >%"
          width %<= roadsWidth >%
```
Spacing is not consistent, I would rather use spaces before and after the expression

Reply:
This is a special case as the evaluation is inside of a string, that's why there are no spaces or they'd show up in the name as well. Either way I changed it.